### PR TITLE
v8 9.6.180.21

### PR DESCRIPTION
**Pull request by iMichka**
Go back to old version as we shipped a beta version by mistake
See https://github.com/Homebrew/homebrew-core/pull/91497

- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
Can we catch this with a livecheck?

**Review by SMillerDev — DISMISSED** [review]

**Comment by samford** [comment]
> Can we catch this with a livecheck?

livecheck was giving the correct stable version (9.6.180.21) at the time the PR for 9.7.106.18 was created/approved (i.e., `v8 : 9.7.106.18 ==> 9.6.180.21`) but having a newer formula version than the livecheck version isn't an error condition, so it's not a CI failure.

I'm a bit wary of treating a newer formula version as an error, as we have some formulae that are temporarily using an unstable version for technical reasons (e.g., `clisp`) and will return to a stable version when there's an appropriate release. In that scenario, we live with the temporary version mismatch as it makes sense for `livecheck` to continue to check for stable versions because we want to know when a newer stable version is released (e.g., 2.49.92 is unstable but we want to know when a stable 2.50 version is released).

[We do have a small number of `livecheck` blocks that match unstable upstream versions to correspond with an unstable formula version but we only do this for formulae that have been (or will be) on an unstable version for a long period of time (e.g., there hasn't been a suitable stable release for years and there's no sign this will change).]

That said, maybe it would make sense to surface this type of livecheck issue as a warning on GitHub but still allow CI to pass. test-bot was updated relatively recently to do this in some other areas, so it may make sense in this context as well. I could work on this if it would be helpful.

**Comment by SMillerDev** [comment]
I really like this warning for casks so I'd be very happy with a similar thing for core. I'm not sure what would be easier, but an automated prerelease label would be a good alternative for me.

**Comment by Bo98** [comment]
FWIW 9.7 will be stable on Jan 4th.

**Comment by samford** [comment]
> I really like this warning for casks so I'd be very happy with a similar thing for core.

I'm not very familiar with cask audits and CI. Could you explain how this works over there or is there an example cask PR that you could point me to? Also, is this different than the prerelease-checking functionality (i.e., checking a GitHub/GitLab tag to make sure it's not marked as pre-release) that homebrew-core and homebrew-cask seem to share?

> I'm not sure what would be easier, but an automated prerelease label would be a good alternative for me.

Outside of GitHub/GitLab releases that are clearly marked as prerelease, automatically identifying prerelease versions can be tricky and/or error-prone. In some cases, we would need to understand the version scheme to know that a version isn't stable (e.g., unstable versions being indicated by an odd-numbered minor version or a patch version above 90) and I don't believe this is broadly feasible (i.e., few projects explicitly specify their version scheme, some projects frustratingly stray from their apparent version scheme at times, etc.). When it comes to this particular `v8` situation, it can only be identified by checking upstream information (as livecheck does).

Simply from the perspective of livecheck, the `newer_than_upstream` boolean (i.e., the formula version is newer than the livecheck version) being `true` doesn't necessarily indicate a prerelease version. I do use this information to identify formulae that have been bumped to an inappropriate version (as here) but it can also mean that a contributor/maintainer has independently identified a new version that livecheck doesn't and we need to modify/add a `livecheck` block to address the mismatch. I've probably seen an equal share of both situations in practice.

If we want to work with this indicator from livecheck, I think it makes sense to simply surface it as a GitHub warning (e.g., "Formula version (9.7.106.18) is newer than livecheck version (9.6.180.21)"), so a human can determine if it's actionable and what needs to be done. Automatically inferring anything beyond this (e.g., the formula version is a prerelease version) is only going to be correct part of the time, so it wouldn't be a reliable indicator.

**Comment by carlocab** [comment]
> FWIW 9.7 will be stable on Jan 4th.

In that case I'd favour waiting rather than creating lots of extra churn for users who already have this installed.

**Comment by iMichka** [comment]
BTW there seem to be an issue with the bottle upload when there is a version scheme bump.

**Review by carlocab — CHANGES_REQUESTED** [review]
This will be stable soon, so let's just wait (especially since @BrewTestBot doesn't seem keen on publishing this either)

**Comment by carlocab** [comment]
> BTW there seem to be an issue with the bottle upload when there is a version scheme bump. 

Yes, I think that's because we use versions to identify bottles on GitHub Packages, but version scheme changes are not visible in the version.

**Comment by iMichka** [comment]
Marking this as "do not merge", will re-check this on Jan 4th then.

**Comment by SMillerDev** [comment]
> I think bayandin encountered this previously. Not sure how they got around it. (Please don't delete bottles that users have already had access to.)

You could do a revision bump

**Comment by carlocab** [comment]
> > I think bayandin encountered this previously. Not sure how they got around it. (Please don't delete bottles that users have already had access to.)
> 
> You could do a revision bump

You'd have to do this for 9.7.106.18 too if that version ever needs publishing again tho

**Comment by chenrui333** [comment]
The stable v8 is `9.7.106.18` now

**Comment by carlocab** [comment]
The current version is stable now. Let's be more careful about bumping the version of this formula in the future.

**Pull request closed without merging**
